### PR TITLE
Add YAML description meta tags

### LINKED ISSUE #2573: Add description to YAML front matter to improve SEO rankings
As suggested by @qdequele, we might be able to make the documentation easier to find for some users by writing summaries for all/the majority of the content in the docs.

### PR DESCRIPTION
**Pull request by guimachiavelli**
Fixes #2573

### CODE REVIEW

File: learn/cookbooks/postman_collection.mdx [hunk 1]
```diff
@@ -1,3 +1,7 @@
+---
+description: Postman is a platform that lets you create HTTP requests you can easily reuse and share with everyone. This how-to guide explains how to use Postman to help testing and debugging Meilisearch's API.
```
Comment (guimachiavelli) [inline learn/cookbooks/postman_collection.mdx:2]:
```suggestion
description: This how-to guide explains how to use Postman when testing and debugging Meilisearch's API.
```

File: learn/cookbooks/search_bar_for_docs.mdx [hunk 2]
```diff
@@ -1,3 +1,7 @@
+---
+description: Use Meilisearch to index content in a text-heavy website. Covers installing Meilisarch, configuring a text scraper, and creating a simple front end.
```
Comment (guimachiavelli) [inline learn/cookbooks/search_bar_for_docs.mdx:2]:
```suggestion
description: Use Meilisearch to index content in a text-heavy website. Covers installing Meilisearch, configuring a text scraper, and creating a simple front end.
```

File: learn/core_concepts/relevancy.mdx [hunk 3]
```diff
@@ -1,5 +1,7 @@
 ---
 sidebarDepth: 4
+description: Relevancy refers to the accuracy of search results. If search results tend to be appropriate, then they can be considered relevant.
```
Comment (guimachiavelli) [inline learn/core_concepts/relevancy.mdx:3]:
```suggestion
description: Relevancy refers to the accuracy of search results. If search results tend to be appropriate for a given query, then they can be considered relevant.
```

File: learn/data_backup/dumps.mdx [hunk 4]
```diff
@@ -1,3 +1,7 @@
+---
+description: Dumps are data backups containing all data related to a Meilisearch instance. They are often useful when migrating to a new Meiliearch release.
```
Comment (guimachiavelli) [inline learn/data_backup/dumps.mdx:2]:
```suggestion
description: Dumps are data backups containing all data related to a Meilisearch instance. They are often useful when migrating to a new Meilisearch release.
```

File: learn/data_backup/snapshots_vs_dumps.mdx [hunk 5]
```diff
@@ -1,3 +1,7 @@
+---
+description: "Meilisearch offers two types of backups: snapshots and dumps. Snapshots are mainly intended as a safeguard, while dumps are more useful when migrating Meilisearch."
```
Comment (guimachiavelli) [inline learn/data_backup/snapshots_vs_dumps.mdx:2]:
```suggestion
description: "Meilisearch offers two types of backups: snapshots and dumps. Snapshots are mainly intended as a safeguard, while dumps are useful when migrating Meilisearch."
```

File: learn/experimental/metrics.mdx [hunk 6]
```diff
@@ -1,3 +1,7 @@
+---
+description: The /metrics endpoint is an experimental feature. It exposes data compatible with Prometheus and offers valuable insight into Meilisearch's behavior and performance.
```
Comment (guimachiavelli) [inline learn/experimental/metrics.mdx:2]:
```suggestion
description: The /metrics endpoint is an experimental feature. It exposes data compatible with Prometheus and offers insight into Meilisearch's behavior and performance.
```

File: learn/experimental/overview.mdx [hunk 7]
```diff
@@ -1,3 +1,7 @@
+---
+description: Activate and configure Meilisearch experimental features. Meilisearch's experimental features offer exciting new functionality that is still undergoing development.
```
Comment (guimachiavelli) [inline learn/experimental/overview.mdx:2]:
```suggestion
description: Activate and configure Meilisearch experimental features. Meilisearch's experimental features offer exciting new functionality that is still in development.
```

File: learn/experimental/ranking_rule_score_details.mdx [hunk 8]
```diff
@@ -1,3 +1,7 @@
+---
+description: The --experimental-score-details flag offers in-depth data about how a query's ranking scores.
```
Comment (guimachiavelli) [inline learn/experimental/ranking_rule_score_details.mdx:2]:
```suggestion
description: The --experimental-score-details flag offers in-depth data about a query's ranking scores.
```

File: learn/experimental/reduce-indexing-memory-usage.mdx [hunk 9]
```diff
@@ -1,3 +1,7 @@
+---
+description: The --experimental-reduce-indexing-memory-usage enables MDB_WRITEMAP, which might improve performance in some UNIX and UNIX-like setups.
```
Comment (guimachiavelli) [inline learn/experimental/reduce-indexing-memory-usage.mdx:2]:
```suggestion
description: The --experimental-reduce-indexing-memory-usage enables MDB_WRITEMAP, which can improve performance in some UNIX and UNIX-like setups.
```

File: learn/fine_tuning_results/faceted_search.mdx [hunk 10]
```diff
@@ -1,5 +1,6 @@
 ---
 sidebarDepth: 3
+description: Faceted search interfaces are provide users with a quick way to narrow down search results by selecting categories relevant to their query.
```
Comment (guimachiavelli) [inline learn/fine_tuning_results/faceted_search.mdx:3]:
```suggestion
description: Faceted search interfaces provide users with a quick way to narrow down search results by selecting categories relevant to their query.
```

File: learn/fine_tuning_results/filtering.mdx [hunk 11]
```diff
@@ -1,5 +1,6 @@
 ---
 sidebarDepth: 4
+description: Filters allow your users to refine search results and are a fundamental element when creating faceted search interfaces.
```
Comment (guimachiavelli) [inline learn/fine_tuning_results/filtering.mdx:3]:
```suggestion
description: Filters allow your users to refine search results. They can also be used when creating faceted search interfaces.
```

File: learn/fine_tuning_results/working_with_dates.mdx [hunk 12]
```diff
@@ -1,3 +1,7 @@
+---
+description: Learn about Meilisearch's approach to date and time values, how to index documents with chronological data, and how to sort and filter search results based on time.
```
Comment (guimachiavelli) [inline learn/fine_tuning_results/working_with_dates.mdx:2]:
```suggestion
description: Learn how to index documents with chronological data, and how to sort and filter search results based on time.
```

File: learn/getting_started/installation.mdx [hunk 13]
```diff
@@ -1,5 +1,6 @@
 ---
 sidebarDepth: 3
+description: Use Meilisearch with Meilisearch Cloud, another cloud service, or install it locally.
```
Comment (guimachiavelli) [inline learn/getting_started/installation.mdx:3]:
```suggestion
description: Use Meilisearch with either Meilisearch Cloud, another cloud service, or install it locally.
```

File: learn/index_settings/all_index_settings.mdx [hunk 14]
```diff
@@ -1,3 +1,7 @@
+---
+description: Meilisearch has several index-specific settings to ensure maximum relevancy with many different datasets and use-cases.
```
Comment (guimachiavelli) [inline learn/index_settings/all_index_settings.mdx:2]:
```suggestion
description: Meilisearch has several index-specific settings to ensure maximum relevancy with different datasets and use-cases.
```

File: learn/inner_workings/tokenization.mdx [hunk 15]
```diff
@@ -1,3 +1,7 @@
+---
+description: Tokenization is the act of taking a sentence or phrase and splitting it into smaller units of language. It is a crucial process when indexing documents.
```
Comment (guimachiavelli) [inline learn/inner_workings/tokenization.mdx:2]:
```suggestion
description: Tokenization is the process of taking a sentence or phrase and splitting it into smaller units of language. It is a crucial procedure when indexing documents.
```

File: learn/update_and_migration/algolia_migration.mdx [hunk 16]
```diff
@@ -1,5 +1,6 @@
 ---
 sidebarDepth: 3
+description: This guide will take you step-by-step through the creation of a Node.js script to upload Algolia index data to Meilisearch.
```
Comment (guimachiavelli) [inline learn/update_and_migration/algolia_migration.mdx:3]:
```suggestion
description: This guide will take you step-by-step through the creation of a Node.js script to upload data indexed by Algolia to Meilisearch.
```

File: learn/update_and_migration/versioning.mdx [hunk 17]
```diff
@@ -1,3 +1,7 @@
+---
+description: This article describes the system behind Meilisearch's SDK and engine version numbering, as well as compatibility between different releases.
```
Comment (guimachiavelli) [inline learn/update_and_migration/versioning.mdx:2]:
```suggestion
description: This article describes the system behind Meilisearch's SDK and engine version numbering and compatibility.
```

File: reference/api/overview.mdx [hunk 18]
```diff
@@ -1,3 +1,7 @@
+---
+description: Consult this page for an overview of how to query Meilisearch's API, which types of parameters it supports, and how it structures responses.
```
Comment (guimachiavelli) [inline reference/api/overview.mdx:2]:
```suggestion
description: Consult this page for an overview of how to query Meilisearch's API, which types of parameters it supports, and how it structures its responses.
```

**Comment by qdequele** [comment]
Hello @guimachiavelli it's great! Is it also possible to add the titles? I did it in my test PR. https://github.com/meilisearch/documentation/pull/2571/files

**Review by qdequele — CHANGES_REQUESTED** [review]
Hello @guimachiavelli it's great! Is it also possible to add the titles? I did it in my test PR. https://github.com/meilisearch/documentation/pull/2571/files

**Comment by guimachiavelli** [comment]
Hi @qdequele! My plan is to address the titles programmatically, as the majority should really be `[Page's H1] — Meilisearch documentation`. So, for example, the page title for the cloud getting started would be `Getting started with Meilisearch Cloud — Meilisearch documentation`. I'm looking into how to do that, but I believe we might be able to skip basement.

**Comment by guimachiavelli** [comment]
After some research, `<title>`s were not implemented in a way we can easily and safely customise. Skipping basement and doing it ourselves does not look like a good choice.

Taking in consideration the time it will take for basement to do this, though, I have manually edited the `<title>` for each documentation page. 

At a later date, once they're done with our current requests, we'll investigate how we can properly automate this.

**Review by qdequele — APPROVED** [review]
LGTM

**Review by curquiza — APPROVED** [review]

**Pull request merged**